#
# This is to work around an apparent ncurses bug.. It's a slight optimisation
# too, I think.
#
--- Win.c.dist	Wed Dec  6 13:38:47 1995
+++ Win.c	Sun Aug 18 23:08:11 1996
@@ -1101,16 +1101,12 @@
 			Exit(kExitWinFail2);
 	}
 
+	werase(gPromptWin);
 	WAttr(gPromptWin, flags, 1);
-	if (addTail) {
-		mvwprintw(gPromptWin, 0,0, "%s%s", pr, kPromptTail);
-		WAttr(gPromptWin, flags, 0);
-	} else {
-		mvwprintw(gPromptWin, 0,0, "%s", pr);
-		WAttr(gPromptWin, flags, 0);
-		if (flags & kReverse)
-			wprintw(gPromptWin, " ");
-	}
+	mvwprintw(gPromptWin, 0,0, "%s", pr);
+	if (addTail)
+		wprintw(gPromptWin, "%s", kPromptTail);
+	WAttr(gPromptWin, flags, 0);
 	wnoutrefresh(gPromptWin);
 	werase(gInputWin);
 	touchwin(gInputWin);
